Tests do not work on non-x86

```rust
+ /usr/bin/env CARGO_HOME=.cargo RUSTC_BOOTSTRAP=1 /usr/bin/cargo test -j2 -Z avoid-dev-deps --release --no-fail-fast
       Fresh semver-parser v0.7.0
       Fresh stable_deref_trait v1.1.1
       Fresh semver v0.9.0
       Fresh rustc_version v0.2.3
       Fresh libc v0.2.71
   Compiling mbox v0.5.0 (/builddir/build/BUILD/mbox-0.5.0)
     Running `/usr/bin/rustc --crate-name mbox src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --emit=dep-info,link -C opt-level=3 --test --cfg 'feature="default"' --cfg 'feature="std"' -C metadata=90240f85f6eb9d26 -C extra-filename=-90240f85f6eb9d26 --out-dir /builddir/build/BUILD/mbox-0.5.0/target/release/deps -L dependency=/builddir/build/BUILD/mbox-0.5.0/target/release/deps --extern libc=/builddir/build/BUILD/mbox-0.5.0/target/release/deps/liblibc-41bc702636c2763a.rlib --extern stable_deref_trait=/builddir/build/BUILD/mbox-0.5.0/target/release/deps/libstable_deref_trait-c6befc4669bcf70e.rlib -Copt-level=3 -Cdebuginfo=2 -Clink-arg=-Wl,-z,relro,-z,now -Ccodegen-units=1 --cap-lints=warn --cfg stable_channel`
warning: use of deprecated item 'try': use the `?` operator instead
   --> src/sentinel.rs:110:20
    |
110 |         let mbox = try!(MBox::from_raw_utf8_parts(base as *mut u8, len + 1));
    |                    ^^^
    |
    = note: `#[warn(deprecated)]` on by default
warning: use of deprecated item 'try': use the `?` operator instead
   --> src/mbox.rs:730:22
    |
730 |         let string = try!(from_utf8(bytes)) as *const str as *mut str;
    |                      ^^^
warning: 2 warnings emitted
warning: use of deprecated item 'try': use the `?` operator instead
   --> src/sentinel.rs:110:20
    |
110 |         let mbox = try!(MBox::from_raw_utf8_parts(base as *mut u8, len + 1));
    |                    ^^^
    |
    = note: `#[warn(deprecated)]` on by default
warning: use of deprecated item 'try': use the `?` operator instead
   --> src/mbox.rs:730:22
    |
730 |         let string = try!(from_utf8(bytes)) as *const str as *mut str;
    |                      ^^^
error[E0600]: cannot apply unary operator `-` to type `u8`
   --> src/sentinel.rs:319:26
    |
319 |         *src.offset(1) = -0x19;
    |                          ^^^^^ cannot apply unary operator `-`
    |
    = note: unsigned values cannot be negated
error[E0600]: cannot apply unary operator `-` to type `u8`
   --> src/sentinel.rs:320:26
    |
320 |         *src.offset(2) = -0x6c;
    |                          ^^^^^ cannot apply unary operator `-`
    |
    = note: unsigned values cannot be negated
error[E0600]: cannot apply unary operator `-` to type `u8`
   --> src/sentinel.rs:321:26
    |
321 |         *src.offset(3) = -0x4e;
    |                          ^^^^^ cannot apply unary operator `-`
    |
    = note: unsigned values cannot be negated
error[E0600]: cannot apply unary operator `-` to type `u8`
   --> src/sentinel.rs:333:26
    |
333 |         *src.offset(0) = -1;
    |                          ^^ cannot apply unary operator `-`
    |
    = note: unsigned values cannot be negated
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -72,6 +72,10 @@
 #![cfg_attr(nightly_channel, feature(min_specialization, unsize, coerce_unsized))]
 #![cfg_attr(not(feature = "std"), no_std)]
 
+#[cfg(doctest)]
+#[doc = include_str!("../README.md")]
+extern "C" {}
+
 #[cfg(not(feature = "std"))]
 extern crate core as std;
 extern crate libc;

--- a/src/mbox.rs
+++ b/src/mbox.rs
@@ -954,6 +954,12 @@ fn test_string_from_bytes() {
 }
 
 #[test]
+fn test_string_with_internal_nul() {
+    let string = MBox::<str>::from("ab\0c");
+    assert_eq!(&*string, "ab\0c");
+}
+
+#[test]
 fn test_non_utf8() {
     let bytes = MBox::from_slice(b"\x88\x88\x88\x88");
     let string = MBox::from_utf8(bytes);

--- a/src/sentinel.rs
+++ b/src/sentinel.rs
@@ -328,9 +328,9 @@ fn test_string() {
     unsafe {
         let src = gen_malloc::<c_char>(5).as_ptr();
         *src.offset(0) = 0x61;
-        *src.offset(1) = -0x19;
-        *src.offset(2) = -0x6c;
-        *src.offset(3) = -0x4e;
+        *src.offset(1) = -0x19i8 as c_char;
+        *src.offset(2) = -0x6ci8 as c_char;
+        *src.offset(3) = -0x4ei8 as c_char;
         *src.offset(4) = 0;
 
         let string = MString::from_raw_unchecked(src);
@@ -342,7 +342,7 @@ fn test_string() {
 fn test_non_utf8_string() {
     unsafe {
         let src = gen_malloc::<c_char>(2).as_ptr();
-        *src.offset(0) = -1;
+        *src.offset(0) = -1i8 as c_char;
         *src.offset(1) = 0;
 
         let string = MString::from_raw(src);

